packages/react-native-webview/android/src/test/kotlin/com/datadog/reactnative/webview/DatadogWebViewTest.kt

```diff
@@ -0,0 +1,120 @@
+package com.datadog.reactnative.webview
+
+import com.datadog.android.api.SdkCore
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.webview.WebViewTracking
+import com.datadog.reactnative.DatadogSDKWrapperStorage
+import com.datadog.reactnative.tools.unit.GenericAssert.Companion.assertThat
+import com.facebook.react.bridge.JavaOnlyArray
+import com.facebook.react.uimanager.ThemedReactContext
+import com.reactnativecommunity.webview.RNCWebView
+import com.reactnativecommunity.webview.RNCWebViewWrapper
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.MockedStatic
+import org.mockito.Mockito
+import org.mockito.Mockito.mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class DatadogWebViewTest {
+
+    @Mock
+    lateinit var themedReactContext: ThemedReactContext
+
+    @Mock
+    lateinit var datadogCore: InternalSdkCore
+
+    private lateinit var webViewTrackingMockedStatic: MockedStatic<WebViewTracking>
+
+    @BeforeEach
+    fun `set up`() {
+        whenever(themedReactContext.runOnUiQueueThread(any())).thenAnswer { answer ->
+            answer.getArgument<Runnable>(0).run()
+            true
+        }
+
+        webViewTrackingMockedStatic = Mockito.mockStatic(WebViewTracking::class.java)
+        webViewTrackingMockedStatic.`when`<Unit> {
+            WebViewTracking.enable(
+                webView = any(), // Mock the WebView parameter
+                allowedHosts = any(), // Mock the list of allowed hosts
+                logsSampleRate = any(), // Mock the logsSampleRate parameter
+                sdkCore = any() // Mock the SdkCore parameter
+            )
+        }.then {} // Return Unit as the function has no return value
```

Comment:
why do we need to mock this one? just for verification?

Reply:
Yes, to verify that it is eventually called, particularly in the use case where WebView is created before the Datadog core has been initialized.